acceptance-tests/src/test/java/tech/pegasys/ethsigner/tests/hashicorpVault/HashicorpVaultDocker.java

```diff
@@ -0,0 +1,222 @@
+    final ExecStartResultCallback execStartResultCallback =
+        docker.execStartCmd(execCreateCmdResponse.getId()).exec(resultCallback).awaitCompletion();
+    execStartResultCallback.onError(
+        new RuntimeException(
+            "command in Hashicorp Vault returned error\n" + execStartResultCallback.toString()));
+    if (expectedInStdout != null) {
+      final String stdoutString = stdout.toString();
+      return stdoutString.indexOf(expectedInStdout) != -1;
+    } else {
+      return true;
+    }
+  }
+
+  public void shutdown() {
+    if (hasVaultContainer()) {
+      stopVaultContainer();
+      removeVaultContainer();
+    }
+  }
+
+  /** This method now also creates the secret that contains the private signing key. */
+  public void awaitStartupCompletion() {
+    LOG.info("Waiting for Hashicorp Vault to become responsive...");
+    final ExecCreateCmdResponse execCreateCmdResponse =
+        getExecCreateCmdResponse(CREATE_ETHSIGNER_SIGNING_KEY_SECRET);
+    waitFor(
+        () ->
+            assertThat(runCommandInVaultContainer(execCreateCmdResponse, "created_time")).isTrue());
+    LOG.info("Hashicorp Vault is now responsive");
+  }
+
+  public int port() {
+    return port;
+  }
+
+  private boolean hasVaultContainer() {
+    return docker != null && vaultContainerId != null;
+  }
+
+  private void stopVaultContainer() {
+    try {
+      LOG.info("Stopping the Vault Docker container...");
+      docker.stopContainerCmd(vaultContainerId).exec();
+      final WaitContainerResultCallback waiter = new WaitContainerResultCallback();
+      docker.waitContainerCmd(vaultContainerId).exec((waiter));
+      waiter.awaitCompletion();
+      LOG.info("Stopped the Vault Docker container");
+    } catch (final NotModifiedException e) {
+      LOG.error("Vault Docker container has already stopped");
+    } catch (final InterruptedException e) {
+      LOG.error("Interrupted when waiting for Vault Docker container to stop");
+    }
+  }
+
+  private void removeVaultContainer() {
+    LOG.info("Removing the Vault Docker container...");
+    docker.removeContainerCmd(vaultContainerId).withForce(true).exec();
+    LOG.info("Removed the Vault Docker container");
+  }
+
+  private void pullVaultImage() {
+    final PullImageResultCallback callback = new PullImageResultCallback();
+    docker.pullImageCmd(HASHICORP_VAULT_IMAGE).exec(callback);
+
+    try {
+      LOG.info("Pulling the Vault Docker image...");
+      callback.awaitCompletion();
+      LOG.info("Pulled the Vault Docker image: " + HASHICORP_VAULT_IMAGE);
+    } catch (final InterruptedException e) {
+      LOG.error(e);
+    }
+  }
+
+  private String createVaultContainer() {
+    final HostConfig hostConfig =
+        HostConfig.newHostConfig()
+            .withPortBindings(httpPortBinding())
+            .withCapAdd(Capability.IPC_LOCK);
+
+    final List<String> enviromentVars = new ArrayList<String>();
```

Comment:
nit: can use new ArrayList<>() to simplify this

Reply:
Done